### PR TITLE
Add --node-provisioning-mode flag

### PR DESCRIPTION
**Pull request by matthchr**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

**Review by FumingZhang — COMMENTED** [review]
lgtm

please add a live test case for the newly added options, may find examples from `test_aks_commands.py`

**Comment by matthchr** [comment]
> please add a live test case for the newly added options, may find examples from test_aks_commands.py

Hi @FumingZhang - right now I cannot add a live test case that passes because the feature is disabled in all regions. I can add a test case which asserts that we get an error, does that work?

I'll update the PR with such a test.

**Comment by matthchr** [comment]
Just realized I can just test `Manual` mode instead. THat works well enough for CLI purposes to ensure that the field is flowing I think.

**Comment by matthchr** [comment]
added test

**Review by Bryce-Soghigian — APPROVED** [review]

**Review by FumingZhang — APPROVED** [review]
LGTM

Queue [live test](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=82072800&view=results) to validate the change.

Please help resolve merge conflict

**Comment by matthchr** [comment]
Updated

File: src/aks-preview/HISTORY.rst [hunk 1]
```diff
@@ -9,9 +9,11 @@ If there is no rush to release a new version, please just add a description of t
 
 To release a new version, please select a new version number (usually plus 1 to last patch version, X.Y.Z -> Major.Minor.Patch, more details in `\doc <https://semver.org/>`_), and then add a new section named as the new version number in this file, the content should include the new modifications and everything from the *Pending* section. Finally, update the `VERSION` variable in `setup.py` with this new version number.
 
-Pending
+0.5.167
```
Comment (zhoxing-ms) [inline src/aks-preview/HISTORY.rst:12]: > 0.5.167

The extension version `0.5.167` already exists, please do not write the duplicate extension version in the release notes

File: src/aks-preview/azext_aks_preview/_help.py [hunk 2]
```diff
@@ -573,6 +573,9 @@
         - name: --enable-cost-analysis
           type: bool
           short-summary: Enable exporting Kubernetes Namespace and Deployment details to the Cost Analysis views in the Azure portal. For more information see aka.ms/aks/docs/cost-analysis.
+        - name: --node-provisioning-mode
+          type: string
+          short-summary: Sets the node provisioning mode of the cluster. Valid values are "Auto" and "Manual". For more information on "Auto" mode see aka.ms/aks/nap.
```
Comment (zhoxing-ms) [inline src/aks-preview/azext_aks_preview/_help.py:578]: Please use the first-person voice in the help message
```suggestion
          short-summary: Set the node provisioning mode of the cluster. Valid values are "Auto" and "Manual". For more information on "Auto" mode see aka.ms/aks/nap.
```

File: src/aks-preview/azext_aks_preview/_help.py [hunk 3]
```diff
@@ -1087,6 +1090,9 @@
         - name: --disable-cost-analysis
           type: bool
           short-summary: Disable exporting Kubernetes Namespace and Deployment details to the Cost Analysis views in the Azure portal.
+        - name: --node-provisioning-mode
+          type: string
+          short-summary: Sets the node provisioning mode of the cluster. Valid values are "Auto" and "Manual". For more information on "Auto" mode see aka.ms/aks/nap.
```
Comment (zhoxing-ms) [inline src/aks-preview/azext_aks_preview/_help.py:1095]: Same as above
```suggestion
          short-summary: Set the node provisioning mode of the cluster. Valid values are "Auto" and "Manual". For more information on "Auto" mode see aka.ms/aks/nap.
```

File: src/aks-preview/azext_aks_preview/_params.py [hunk 4]
```diff
@@ -504,6 +512,8 @@ def load_arguments(self, _):
                    help='set azure disk type storage pool sku for azure container storage')
         c.argument('storage_pool_option', arg_type=get_enum_type(storage_pool_options),
                    help='set ephemeral disk storage pool option for azure container storage')
+        c.argument('node_provisioning_mode', is_preview=True, arg_type=get_enum_type(node_provisioning_modes),
+                   help='Sets the node provisioning mode of the cluster. Valid values are "Auto" and "Manual". For more information on "Auto" mode see aka.ms/aks/nap.')
```
Comment (zhoxing-ms) [inline src/aks-preview/azext_aks_preview/_params.py:516]:
```suggestion
                   help='Set the node provisioning mode of the cluster. Valid values are "Auto" and "Manual". For more information on "Auto" mode see aka.ms/aks/nap.')
```

File: src/aks-preview/azext_aks_preview/_params.py [hunk 5]
```diff
@@ -646,6 +656,8 @@ def load_arguments(self, _):
                    help='set ephemeral disk storage pool option for azure container storage')
         c.argument('azure_container_storage_nodepools',
                    help='define the comma separated nodepool list to install azure container storage')
+        c.argument('node_provisioning_mode', is_preview=True, arg_type=get_enum_type(node_provisioning_modes),
+                   help='Sets the node provisioning mode of the cluster. Valid values are "Auto" and "Manual". For more information on "Auto" mode see aka.ms/aks/nap.')
```
Comment (zhoxing-ms) [inline src/aks-preview/azext_aks_preview/_params.py:660]:
```suggestion
        c.argument('node_provisioning_mode', is_preview=True, arg_type=get_enum_type(node_provisioning_modes),
                   help='Set the node provisioning mode of the cluster. Valid values are "Auto" and "Manual". For more information on "Auto" mode see aka.ms/aks/nap.')
```

**Comment by yanzhudd** [comment]
it is suggested to add test for update command as well

**Comment by FumingZhang** [comment]
Added the test for update command and resolved conflict in PR #6953.

**Pull request closed without merging**